hdfs-dataset-sink/src/main/java/org/springframework/cloud/stream/module/dataset/sink/DatasetSinkProperties.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.dataset.sink;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Used to configure the HDFS Dataset Sink module options that are related to writing using Kite SDK.
+ *
+ * @author Thomas Risberg
+ */
+@SuppressWarnings("unused")
+@ConfigurationProperties
+public class DatasetSinkProperties {
+
+	/**
+	 * The URI to use to access the Hadoop FileSystem.
+	 */
+	private String fsUri;
+
+	/**
+	 * The base directory path where the files will be written in the Hadoop FileSystem.
+	 */
+	private String directory = "/tmp/hdfs-dataset-sink";
+
+	/**
+	 * The sub-directory under the basePath where files will be written.
+	 */
+	@Value("${spring.application.name:data}")
+	private String namespace;
+
+	/**
+	 * Threshold in number of messages when file will be automatically flushed and rolled over.
+	 */
+	private int batchSize = 10000;
+
+	/**
+	 * Idle timeout in milliseconds when Hadoop file resource is automatically closed.
+	 */
+	private long idleTimeout = 0L;
+
+	/**
+	 * Whether null property values are allowed, if set to true then schema will use UNION for each field.
+	 */
+	private boolean allowNullValues = false;
+
+	/**
+	 * The format to use, valid options are avro and parquet.
+	 */
+	private String format = "avro";
+
+	/**
+	 * The partition path strategy to use, a list of KiteSDK partition expressions separated by a '/' symbol.
+	 */
+	private String partitionPath = null;
+
+	/**
+	 * The size of the cache to be used for partition writers (10 if omitted).
```

Comment:
if it's not specified we don't want to set this since that adds a runtime system property just for this rarely used setting, so we need to keep track of when this property is not set